src/main/java/com/puppycrawl/tools/checkstyle/checks/modifier/RedundantModifierCheck.java

```diff
@@ -270,27 +270,9 @@ private void checkInterfaceModifiers(DetailAST ast) {
      */
     private void checkEnumConstructorModifiers(DetailAST ast) {
         final DetailAST modifiers = ast.findFirstToken(TokenTypes.MODIFIERS);
-        final DetailAST modifier = getFirstModifierAst(modifiers);
-
-        if (modifier != null) {
-            log(modifier, MSG_KEY, modifier.getText());
-        }
-    }
-
-    /**
-     * Retrieves the first modifier that is not an annotation.
-     *
-     * @param modifiers The ast to examine.
-     * @return The first modifier or {@code null} if none found.
-     */
-    private static DetailAST getFirstModifierAst(DetailAST modifiers) {
-        DetailAST modifier = modifiers.getFirstChild();
-
-        while (modifier != null && modifier.getType() == TokenTypes.ANNOTATION) {
-            modifier = modifier.getNextSibling();
-        }
-
-        return modifier;
+        TokenUtil.findFirstTokenByPredicate(
+            modifiers, mod -> mod.getType() != TokenTypes.ANNOTATION
+        ).ifPresent(modifier -> log(modifier, MSG_KEY, modifier.getText()));
```

Comment:
I can't say for sure, but I don't like this. I would like to see the closing parenthesis on the previous line. There is a chain, but no indentation.

Reply:
```
        TokenUtil.findFirstTokenByPredicate(
                modifiers, mod -> mod.getType() != TokenTypes.ANNOTATION)
            .ifPresent(modifier -> log(modifier, MSG_KEY, modifier.getText()));
```

I do not have strong opinion on this for now.